internal/txvalidation/validator.go

```diff
@@ -91,20 +91,53 @@ func (v *Validator) ValidateBlock(block *types.Block) ([]*types.ValidationInfo,
 			return nil, err
 		}
 
+		if err = v.dataTxValidator.parallelValidation(dataTxEnvs, usersWithValidSigPerTX, valInfoArray); err != nil {
+			return nil, errors.WithMessage(err, "error while validating data transaction")
+		}
+
 		pendingOps := newPendingOperations()
 		for txNum, txEnv := range dataTxEnvs {
 			if valInfoArray[txNum].Flag != types.Flag_VALID {
 				continue
 			}
 
-			valRes, err := v.dataTxValidator.validate(txEnv, usersWithValidSigPerTX[txNum], pendingOps)
-			if err != nil {
-				return nil, errors.WithMessage(err, "error while validating data transaction")
-			}
+			invalid := false
+			for _, txOps := range txEnv.Payload.DbOperations {
+				var deleteFieldValResult, mvccValResult *types.ValidationInfo
+				var deleteFieldValError, mvccValError error
+
+				var wg sync.WaitGroup
+				wg.Add(2)
+
+				go func() {
+					defer wg.Done()
+					deleteFieldValResult, deleteFieldValError = v.dataTxValidator.validateFieldsInDataDeletes(txOps.DbName, txOps.DataDeletes, pendingOps)
+				}()
+
+				go func() {
+					defer wg.Done()
+					mvccValResult, mvccValError = v.dataTxValidator.mvccValidation(txOps.DbName, txOps, pendingOps)
+				}()
+
+				wg.Wait()
 
-			valInfoArray[txNum] = valRes
-			if valRes.Flag != types.Flag_VALID {
-				v.logger.Debugf("data transaction [%v] is invalid due to [%s]", txEnv.Payload, valRes.ReasonIfInvalid)
+				if deleteFieldValError != nil || mvccValError != nil {
+					return nil, err
+				}
```

Comment:
This returns the wrong error. It should return either `deleteFieldValError` or `mvccValError`.

Reply:
Done